File: iOS/Shiba/Shiba/Types/ShibaToken.swift
//
// Created by 高垣朝陽 on 2018/8/14.
// Copyright (c) 2018 Tlaster. All rights reserved.
//

import Foundation

public class ShibaToken {
    public let prefix: String
    public let value: String

    init(prefix: String, value: String) {
        self.prefix = prefix
        self.value = value
    }

    func isCurrentPlatform(value: String) -> Bool {
        return isCurrentPlatform() && value == self.value;
    }

    func isCurrentPlatform() -> Bool {
        return prefix.isEmpty || prefix == Shiba.instance.configuration.platformType
    }

}

File: Android/shiba/src/main/java/moe/tlaster/shiba/type/ShibaToken.kt
package moe.tlaster.shiba.type

import moe.tlaster.shiba.Shiba
//
//data class ShibaToken(val prefix: String, val value: String) {
//    override fun toString(): String {
//        return if (prefix.isEmpty()) value else "$prefix:$value"
//    }
//
//    fun isCurrentPlatformAndCheckValue(value: String): Boolean {
//        return isCurrentPlatform() && value == this.value
//    }
//
//    fun isCurrentPlatform(): Boolean {
//        return (prefix == Shiba.configuration.platformType || prefix.isEmpty())
//    }
//}